lib/charms/loki_k8s/v1/loki_push_api.py

```diff
@@ -2475,6 +2475,9 @@ def disable_inactive_endpoints(
         container: Container, active_endpoints: Dict[str, str], topology: JujuTopology
     ):
         """Disable forwarding for inactive endpoints by checking against the Pebble plan."""
+        if not container.can_connect():
+            return
```

Comment:
- `_update_endpoints` is called by `_on_pebble_ready` and `_update_logging`.
- `_update_logging` is called by a bunch of relation observers.

Since we don't want to have a can_connect guard in _on_pebble_ready, seems like the best place to move the guard to is `_update_logging`.

I.e. I would remove the can_connect guards from the `_enable_...`/`_disable_...` methods, and keep it "at the top", because `_update_logging` has better context to decide what to do if we can't connect.

Wdyt?

Reply:
Since `can_connect()` is a point-in-time check I tend to use that guard as close as possible to the action that connects through Pebble.
A few milliseconds after `can_connect()` returns True, Pebble may not be available... but, how likely is that possibility? 🤔 

While using `can_connect()` we are in a non-deterministic zone, that's why I believe we need to revisit the [interactions with Pebble](https://github.com/canonical/loki-k8s-operator/issues/434) in all our charms and libs, not just Loki.

BUT... in this specific situation we can move the `can_connect()` guard inside the `for` loop in `_update_logging`, and if it not possible to connect just `continue`... because when Pebble is ready again an event will be fired and we'll end up executing `_update_endpoints` anyway.